docs/MostUsedConditions.md

```diff
@@ -0,0 +1,259 @@
+- `--min-occurrences` for limiting the number of conditions and just show conditions with more occurrences in source-code than treshold. For example `--min-occurrences=5` just will ignore all conditions with four or less times in your source code
+- `--with-flip-check` enables flip check, see [Flip-Check](#flip-check)
+- `--without-occurrences` will hide all occurrences of the conditions and just print the table without any
+- `--without-flags` will hide the "flags" like `flipped` when `--with-flip-check` is enabled
+- `--without-affected-by-processor` will hide the information which processors post-processed a conditions
+- `--with-csv` will export the result table to a csv (comma-separated). Add a filepath as value (absolute or relative) and make sure you have write rights with the current user. Example `--with-csv=output.csv` will create a output.csv file in the current working directory
+- `--csv-delimiter-semicolon` will change the delimiter from the csv to semicolon (;)
+
+### Visitors
+For example we have some code like [this](examples/MostUsedConditions/visitors.php):
+```php
+<?php
+if(date('Y') === 2019) {
+    echo '2019!!!';
+} elseif (date('Y') === 2018) {
+    echo '2018!!!';
+} elseif (date('Y') === 2017) {
+    echo '2017!!!';
+}
+
+if(   date("Y") === 2019) {
+    // do something
+}
+
+$page = isset($_GET['page']) ? $_GET['page'] : 1;
+
+$page = $_GET['page'] ?? 1;
+
+function isAnonymous(?User $user): bool
+{
+    return null === $user;
+}
+```
+The following visitors are available:
+- `If`: Will add all conditions inside if-statements, here it will just add `date('Y') === 2019` twice
+- `ElseIf`: Will add all conditions inside elseif-statements, here it will add `date('Y') === 2018` and <br/>`date('Y') === 2017`
+- `Ternary`: This visitor will add all conditions on the left side of a ternary, here it will add `isset($_GET['page'])`
+- `Coalesce`: Like Ternary, left side will be added, here `$_GET['page']`
+- `BooleanReturn`: (experimental) Will add the return statements without `return` in methods that have :bool as return type (not docblock), here `null === $user`
+
+**Important:** This tool uses a php-parser, and ignore different code-styles. So it's no matter if you are using single-quotes or double-quotes, add different types or amount of whitespaces and so on.
+
+You can combine visitors, use all or just one with a comma-separated list, like `--visitors=If,ElseIf,Ternary` or `--visitors=Coalesce`.
+
+If you mistype a name of a visitor, you will raise an Exception including all possible names (case-sensitive).
+
+<img src="./images/MostUsedConditions/visitorexception.png">
+
+### Processors
+
+You can combine processors, just add them to a comma-seperated list, like <br>`--processors=SplitIsset,SplitLogicalOperator`. The order of processors is important, in the current case it will start with the SplitIsset and then the SplitLogicalOperator. Maybe your results will change if you change the order of the processors.
```

Comment:
```suggestion
You can combine processors. Just add them to a comma-seperated list, like <br>`--processors=SplitIsset,SplitLogicalOperator`. The order of processors is important. In the current case it will start with the SplitIsset and then the SplitLogicalOperator. Your results might change if you change the order of the processors.
```